examples/rust-sdk/lp-bot/src/utils.rs

```diff
@@ -0,0 +1,329 @@
+use clap::ValueEnum;
+use orca_whirlpools::close_position_instructions;
+use orca_whirlpools_client::{Position, Whirlpool};
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::compute_budget::ComputeBudgetInstruction;
+use solana_sdk::{
+    message::Message, program_pack::Pack, pubkey::Pubkey, signature::Signature, signer::Signer,
+    transaction::Transaction,
+};
+use spl_token_2022::state::Mint;
+use std::future::Future;
+use std::str::FromStr;
+use tokio::time::sleep;
+use tokio::time::Duration;
+
+const MAX_RETRIES: usize = 3;
+const INITIAL_RETRY_DELAY: Duration = Duration::from_millis(100);
+
+pub async fn display_position_balances(
+    rpc: &RpcClient,
+    position: &Position,
+    token_mint_a_address: &Pubkey,
+    token_mint_b_address: &Pubkey,
+    decimals_a: u8,
+    decimals_b: u8,
+    slippage_tolerance_bps: u16,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let close_position_instructions = close_position_instructions(
+        &rpc,
+        position.position_mint,
+        Some(slippage_tolerance_bps),
+        None,
+    )
+    .await?;
+
+    let positon_balance_token_a =
+        close_position_instructions.quote.token_est_a as f64 / 10u64.pow(decimals_a as u32) as f64;
```

Comment:
Float conversions like these lose precision. You might be able to do something with a bitshift that has more precision

Reply:
But since it is just for display and not further calculation it is not that big of a deal